pgx-macros/src/lib.rs

```diff
@@ -772,13 +772,13 @@ fn impl_postgres_type(ast: DeriveInput) -> proc_macro2::TokenStream {
             impl #generics JsonInOutFuncs #inout_generics for #name #generics {}
 
             #[doc(hidden)]
-            #[pg_extern(immutable,parallel_safe)]
+            #[pg_extern(immutable,parallel_safe,strict)]
             pub fn #funcname_in #generics(input: &#lifetime pgx::cstr_core::CStr) -> #name #generics {
                 #name::input(input)
             }
```

Comment:
Are we sure these are not going to be Options?

Reply:
I think there are a couple of ways this could work:
* Functions are strict, return a concrete value, and `panic!` if they can't return a value
* Functions are non-strict, return a concrete value, and `panic!` if they can't return a value
* Functions are non-strict, return an `Option`, and `error!` or `warn!` or whatever if they decide they want to, and just return `None` sometimes.
* Functions are non-strict, and return a `Result` (or even `Result<Option>`) which is somehow transmuted into a `panic!` in the `Err` case.

In the first case, `NULL` always translates to `NULL` as a custom type. I think this is reasonable, but maybe there's some domain where `NULL` should translate to something else: an empty string, 0, banana flavor. If we think we need to support that case, then this option won't work.

In the second, everyone has to handle `Option`s, but hey it's not too bad. This doesn't expose a way of explicitly returning `NULL` from non-null input text.

The third is probably my favorite of the non-strict options. Explicit handling of input and output NULLs, panic or error or whatever if you want to.

The fourth is maybe more rust-friendly, if this function is going to be called from non-pgx code. Something like this (and similar for `PgVarlenaInOutFuncs` and `JsonInOutFuncs`):
```rust
pub trait InOutFuncs<E> {
    fn input(input: &Option<crate::cstr_core::CStr>) -> Result<Option<Self>, E>
    where
        Self: Sized;
    fn output(&self, buffer: &mut StringInfo);
}
```
This would need a small shim to unwrap the result, but shouldn't be too bad.